File: trunk/eg/src/main/java/net/esper/example/marketdatafeed/RateReportingListener.java
package net.esper.example.marketdatafeed;

import net.esper.client.UpdateListener;
import net.esper.event.EventBean;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

public class RateReportingListener implements UpdateListener
{
    public void update(EventBean[] newEvents, EventBean[] oldEvents)
    {
        logRate(newEvents[0]);
        logRate(newEvents[1]);
    }

    private void logRate(EventBean event)
    {
        log.info("Current rate for feed " + event.get("feed").toString() +
                  " is " + event.get("cnt"));
    }

    private static final Log log = LogFactory.getLog(RateReportingListener.class);
}

File: Examples/example/marketdatafeed/RateReportingListener.cs
using System;

using net.esper.client;
using net.esper.events;

using org.apache.commons.logging;

namespace net.esper.example.marketdatafeed
{
    public class RateReportingListener
    {
        public void Update(EventBean[] newEvents, EventBean[] oldEvents)
        {
            LogRate(newEvents[0]);
            LogRate(newEvents[1]);
        }

        private void LogRate(EventBean eventBean)
        {
            log.Info("Current rate for feed " + eventBean["feed"].ToString() +
                      " is " + eventBean["cnt"]);
        }

        private static readonly Log log = LogFactory.GetLog(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
    }
}